app/controllers/api/v2/app_groups_controller.rb

```diff
@@ -166,25 +166,48 @@ def update_cost
       affected_app: affected_app
     }, status: :ok and return
   end
+
   def deactivated_by_cluster_name
-    @helm_infrastructure = HelmInfrastructure.find_by(cluster_name: params[:cluster_name])
-    
+    cluster_name = params[:cluster_name]
+    app_group_name = params[:name]
+  
+    # Validate presence of both cluster_name and app_group_name
+    unless cluster_name.present? && app_group_name.present?
+      render(json: {
+        success: false,
+        errors: ['Both cluster_name and app_group_name are required'],
+        code: 400,
+      }, status: :bad_request) && return
+    end
+  
+    @helm_infrastructure = HelmInfrastructure.find_by(cluster_name: cluster_name)
+  
     if @helm_infrastructure.blank? || !@helm_infrastructure.active?
       render(json: {
-                 success: false,
-                 errors: ['Helm Infrastructure not found'],
-                 code: 404,
-               }, status: :not_found) && return
+        success: false,
+        errors: ['Helm Infrastructure not found or not active'],
+        code: 404,
+      }, status: :not_found) && return
     end
-    
+  
+    # Check if the Helm Infrastructure's app_group_name matches the provided app_group_name
+    unless @helm_infrastructure.app_group.name == app_group_name
+      render(json: {
+        success: false,
+        errors: ['Mismatched app_group_name and cluster_name'],
+        code: 400,
+      }, status: :bad_request) && return
+    end
+  
     app_group = @helm_infrastructure.app_group
-    barito_apps = app_group.barito_apps
+    barito_apps = app_group.barito_apps 
     barito_apps.each do |app|
       app.update_status('INACTIVE') if app.status == BaritoApp.statuses[:active]
     end
+  
     @helm_infrastructure.update_provisioning_status('DELETE_STARTED')
```

Comment:
do we need to have this? we already have this one, right?
https://github.com/BaritoLog/BaritoMarket/blob/ec9e93ef5b9cde6b0f3d0e2fdafbaa9e92dc8c86/app/workers/delete_helm_infrastructure_worker.rb#L33

Reply:
yes Jun, it can be used as a flag on the test.

Reply:
where in the test that this one is used @fadlinurhasan ? if it's not use anywhere we can remove this one